src/pudl_archiver/archivers/eia/eia176.py

```diff
@@ -41,7 +42,11 @@ async def get_items(self, url: str = items_url) -> list[str]:
     async def get_resources(self) -> ArchiveAwaitable:
         """Download EIA 176 resources."""
         items_list = await self.get_items(url=self.items_url)
-        for year in range(1997, 2023):
+        datasets_list = await self.get_datasets(url=self.base_url, form=self.form)
+        dataset_years = set()
+        for dataset in datasets_list:
+            dataset_years.update([year.ayear for year in dataset.available_years])
```

Comment:
So this is getting a list of available years for the dataset from the API?  At first glance I thought maybe it was looking at the years specified in the data source metadata "available years".  A comment or note in the docstring would help avoid that confusion.

Reply:
Noted, I've added a clarifying string.